celery/worker/consumer/consumer.py

```diff
@@ -696,7 +697,10 @@ def on_task_received(message):
 
     def _restore_prefetch_count_after_connection_restart(self, p, *args):
         with self.qos._mutex:
-            if self._maximum_prefetch_restored:
+            if any([
+                not self.app.conf.worker_enable_prefetch_count,
+                self._maximum_prefetch_restored,
+            ]):
```

Comment:
```suggestion
            if any((
                not self.app.conf.worker_enable_prefetch_count,
                self._maximum_prefetch_restored,
            )):
```
Use a tuple as this is more efficient to initialize.